### PR TITLE
Fix accelerators not being considered when counting allocatables

### PR DESCRIPTION
**Pull request by ElijahQuinones**
**Is this a bug fix or adding new feature?**
bug fix fixes https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/2105
**What is this PR about? / Why do we need it?**
This PR fixes GH https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/2105 as before we where not taking accelerators into consideration when calculating volume limits  
**What testing is done?** 
I spun up and hand added ebs volumes to an instance in each accelerator family to ensure there where no special cases outside of the VT1 family where accelerators take up two slots. 

Additionally I found that the calculation in node.go was incorrect for any instance type in the maxVolumeLimits table the documentation states that for the instance types in that table the calculation of 28 - enis - NVMeInstanceStoreVolumes - gpus/accekerators should be completely  ignored as they are [special cases](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html) . As such I checked with a inf1.xlarge instance type and verified you can in fact attach 26 volumes as it's eni (or anything else for that matter) should not be considered when looking at its limit.

### CODE REVIEW

File: pkg/driver/node.go
Comment (ConnorJC3) [inline pkg/driver/node.go:796]: Why would we not want to subtract the ENIs in this case?

File: pkg/driver/node.go
Comment (ConnorJC3) [inline pkg/driver/node.go:784]: np: capitalize `volume` in variable name

File: hack/generate-gpu-count-table.sh
Comment (ConnorJC3) [inline hack/generate-gpu-count-table.sh:38]:
```suggestion
get_all_gpus | sort | uniq | grep -v "g5.48xlarge"
```

Is this slightly simpler version possible instead? (ditto with other file)

File: pkg/driver/node.go
Comment (ConnorJC3) [inline pkg/driver/node.go:796]:
```suggestion
			availableAttachments = availableAttachments - (enis - 1) - reservedSlots
```

I'm pretty sure this is what we actually should do here, as despite being special cases these devices still are a shared limit.

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [inline pkg/cloud/volume_limits.go:70]: Are we sure f1 is not nitro instance family? 

I see it in the accelerated computing current gen instance list: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances

But perhaps I'm looking in wrong place

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [inline pkg/cloud/volume_limits.go:560]: @ConnorJC3 @torredil Is there a reason we are doing `// /` for comments in this file? 

If not, can we delete all the extra `/`'s?

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [inline pkg/cloud/volume_limits.go:562]: Typo: purposley

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [inline pkg/cloud/volume_limits.go:561]: Typo: excpetion

File: pkg/driver/node.go
Comment (ElijahQuinones) [reply pkg/driver/node.go:796]: Ah yes this is better as if an instance has more than one eni it does count towards the shared limit will fix on next rev thank you !

File: pkg/cloud/volume_limits.go
Comment (ElijahQuinones) [reply pkg/cloud/volume_limits.go:70]: if you run `❯ aws ec2 describe-instance-types --instance-types  f1.2xlarge` you will see that `            "Hypervisor": "xen",` and not nitro. As such I manually confirmed this by attaching 40 volumes to the  f1.2xlarge instance type

File: hack/generate-gpu-count-table.sh
Comment (ElijahQuinones) [reply hack/generate-gpu-count-table.sh:38]:
```
❯ vim test.txt
❯ cat test.txt
"g5.48xlarge": 8,
"g5.43xlarge": 8,
❯ grep -v "(g5.48xlarge)" test.txt
"g5.48xlarge": 8,
"g5.43xlarge": 8,
❯ grep -vE "(g5.48xlarge)" test.txt
"g5.43xlarge": 8,
```

unfortunately no

File: pkg/cloud/volume_limits.go
Comment (ConnorJC3) [reply pkg/cloud/volume_limits.go:560]: No special reason I'm aware of, if we get rid of them preferably do the whole file for consistency.

File: pkg/cloud/volume_limits.go
Comment (ConnorJC3) [reply pkg/cloud/volume_limits.go:70]: I can confirm F1 definitely is non-nitro

**Review by ConnorJC3 — APPROVED** [review]
/lgtm

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [reply pkg/cloud/volume_limits.go:560]: Agreed on delete for whole file

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [reply pkg/cloud/volume_limits.go:70]: Thanks for the deep dive!

File: pkg/driver/node_test.go
Comment (AndrewSirenko) [inline pkg/driver/node_test.go:1235]: Hyper-nit: Add space between `limit` and `(` like we do for other unit tests

Also consider rephrasing parenthetical to `(Instance has attached GPUs and NVMe Instance Store volumes but should be ignored for EBS volume limits calculation)` to clarify the different devices/limits at play here for future contributors. 

Same thing for rest of new unit tests.

**Review by AndrewSirenko — APPROVED** [review]

**Review by ConnorJC3 — APPROVED** [review]
/lgtm

File: pkg/driver/node_test.go
Comment (torredil) [inline pkg/driver/node_test.go:1252]: np: remove 'will and'. Similar comment for other instances of this string such as 1269, 1286, and 1303.

File: pkg/cloud/volume_limits.go
Comment (torredil) [inline pkg/cloud/volume_limits.go:563]: Can we add *where* this information is sourced from in the comment above? it will be helpful to future maintainers.

File: pkg/cloud/volume_limits.go
Comment (torredil) [inline pkg/cloud/volume_limits.go:561]: np: `acclerators` -> `accelerators`

**Review by torredil — APPROVED** [review]
Largely lgtm, thanks @ElijahQuinones

File: pkg/cloud/volume_limits.go
Comment (AndrewSirenko) [reply pkg/cloud/volume_limits.go:563]: +1, especially since AWS Docs are usually really great about re-directing dead pages to their new homes.

File: pkg/cloud/volume_limits.go
Comment (ConnorJC3) [inline pkg/cloud/volume_limits.go:561]: Latest rev adds a typo here: "acclerators"

Also, we should squash before merge. Otherwise lgtm.

**Review by ConnorJC3 — APPROVED** [review]
/lgtm

**Pull request merged**
